x-pack/plugins/alerting_builtins/server/alert_types/geo_threshold/alert_type.ts

```diff
@@ -0,0 +1,173 @@
+import { BUILT_IN_ALERTS_FEATURE_ID } from '../../../common';
+import { getGeoThresholdExecutor } from './geo_threshold';
+import {
+  ActionGroup,
+  AlertServices,
+  ActionVariable,
+  AlertTypeState,
+} from '../../../../alerts/server';
+
+export const GEO_THRESHOLD_ID = '.geo-threshold';
+export type TrackingEvent = 'entered' | 'exited';
+export type BoundaryType = 'entireIndex'; // Will expand to cover more, i.e. - providedShapes, etc.
+export const ActionGroupId = 'tracking threshold met';
+
+const actionVariableContextCrossingEventTimeStampLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextEnteredTimeStampLabel',
+  {
+    defaultMessage: `The time the entity's crossing was recorded to arrive at its current location`,
+  }
+);
+
+const actionVariableContextCurrentLocationLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextCurrentLocationLabel',
+  {
+    defaultMessage: 'The most recently captured location of the entity',
+  }
+);
+
+const actionVariableContextPreviousLocationLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextPreviousLocationLabel',
+  {
+    defaultMessage: 'The previously captured location of the entity',
+  }
+);
+
+const actionVariableContextCurrentBoundaryIdLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextCurrentBoundaryIdLabel',
+  {
+    defaultMessage: 'The current boundary id containing the entity (if any)',
+  }
+);
+
+const actionVariableContextPreviousBoundaryIdLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextPreviousBoundaryIdLabel',
+  {
+    defaultMessage: 'The previous boundary id containing the entity (if any)',
+  }
+);
+
+const actionVariableContextCrossingDocumentIdLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextCrossingDocumentIdLabel',
+  {
+    defaultMessage: 'The id of the crossing entity document',
+  }
+);
+
+const actionVariableContextTimeOfDetectionLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextTimeOfDetectionLabel',
+  {
+    defaultMessage: 'The alert interval end time this change was recorded',
+  }
+);
+
+const actionVariables = {
+  context: [
+    {
+      name: 'crossingEventTimeStamp',
+      description: actionVariableContextCrossingEventTimeStampLabel,
+    },
+    { name: 'currentLocation', description: actionVariableContextCurrentLocationLabel },
+    { name: 'currentBoundaryId', description: actionVariableContextCurrentBoundaryIdLabel },
+    { name: 'previousLocation', description: actionVariableContextPreviousLocationLabel },
+    { name: 'previousBoundaryId', description: actionVariableContextPreviousBoundaryIdLabel },
+    { name: 'crossingDocumentId', description: actionVariableContextCrossingDocumentIdLabel },
+    { name: 'timeOfDetection', description: actionVariableContextTimeOfDetectionLabel },
```

Comment:
After trying these out for a while, and after the changes to the alertinstanceid which is now "entity+boundary", I think these are too limiting.


- In the alerting-app, it's not possible to format a usable message. e.g. a user can't format a message that says `Bus "x" entered region "y" coming from "z"`.
- in the Maps-app, it's not possible to create a tooltip that would filter down on the entity. This makes it really hard to debug. ie. you cannot select a crossing-event, and show the corresponding track by filtering on entity-id / you cannot label the event /...
- It's not possible to display this accurately as a linestring (see earlier comment https://github.com/elastic/kibana/pull/76285#pullrequestreview-497741560). Without it, we cannot map the crossing event as a line (it also makes it hard to conceptually relate the two documents that triggered the detection).

------------

I would also rename "current" and "previous", since these are somewhat confusing. "Current" does not remain "current". It's not "current", in the sense that the crossing-event is triggered after the fact.

---------------

So imho we need to expand on these and add more information:

```
    // Corresponds to the alert
    { name: 'entityId', description: "entityid of the name of the boundary " },
    { name: 'timeOfDetection', description: actionVariableContextTimeOfDetectionLabel },
    { name: 'crossingLine', description: "geojon linestring of the track which connects from-to " },

    // Corresponds to a specific document in the entity-index
    { name: 'toEntityLocation', description: "lon/lat of the entity after it crossed into the boundary"},
    { name: 'toEntityDateTime', description: "timestamp of the entity after it crossed into the boundary"},
    { name: 'toEntityDocumentId', description: "document-id in the entity-index"},

       //  Corresponds to a specific document in the boundary-index
    { name: 'toBoundaryId', description: "id of the boundary the vehicle crossed into ("says OUTSIDE if crossing into outside) },
     'toBoundaryName', description: "human readable name of the boundary the vehicle crossed into (optional-based on boundaryname field)" },

       // ... And the same for `from*`
       // not that this will flip the OUTSIDE-placeholder between to/from for exiting/entering
       ....

```